src/tests/Common/CLRTest.Jit.targets

```diff
@@ -365,6 +372,139 @@ set DOTNET_JitPath=%CORE_ROOT%\superpmi-shim-collector.dll
+  IlasmRoundTrip for merged tests
+  ***********************************************************************************************
+  -->
+
+  <Target Name="IlasmRoundTripCommand" AfterTargets="AfterBuild" Condition="'$(IsMergedTestRunnerAssembly)' == 'true'">
+    <PropertyGroup>
+      <InputAssemblyName>$(AssemblyName).dll</InputAssemblyName>
+      <DisassemblyName>IL-RT/$(AssemblyName).il</DisassemblyName>
+      <TargetAssemblyName>IL-RT/$(AssemblyName).dll</TargetAssemblyName>
+      <_IlasmRoundTripScriptText><![CDATA[
+import os
+import shutil
+import subprocess
+import sys
+import glob
+import struct
+import platform
+
+def is_managed_assembly(file):
+    file_size = os.path.getsize(file)
+
+    if file_size < 64:
+        return False
+
+    with open(file, "rb") as f:
+        f.seek(0x3c)
+
+        # PE Header starts @ 0x3C (60). Its a 4 byte header.
+        peHeaderPointer = struct.unpack('I', f.read(4))[0]
+        if peHeaderPointer == 0:
+            peHeaderPointer = 0x80
+
+        # Ensure there is at least enough room for the following structures:
+        #     24 byte PE Signature & Header
+        #     28 byte Standard Fields         (24 bytes for PE32+)
+        #     68 byte NT Fields               (88 bytes for PE32+)
+        # >= 128 byte Data Dictionary Table
+        if peHeaderPointer > (file_size - 256):
+            return False
+
+        # Check the PE signature.  Should equal 'PE\0\0'.
+        f.seek(peHeaderPointer)
+        peHeaderSignature = struct.unpack('I', f.read(4))[0]
+        if peHeaderSignature != 0x00004550:
+            return False
+
+        # skip over the PEHeader fields
+        f.seek(f.tell() + 20)
+
+        PE32 = 0x10b
+        PE32Plus = 0x20b
+
+        # Read PE magic number from Standard Fields to determine format.
+        peFormat = struct.unpack('H', f.read(2))[0]
+        if peFormat != PE32 and peFormat != PE32Plus:
+            return False
+
+        # Read the 15th Data Dictionary RVA field which contains the CLI header RVA.
+        # When this is non-zero then the file contains CLI data otherwise not.
+        dataDictionaryStart = (peHeaderPointer + (232 if peFormat == PE32 else 248)) & 0xffff
+        f.seek(dataDictionaryStart)
+
+        cliHeaderRva = struct.unpack('I', f.read(4))[0]
+        if cliHeaderRva == 0:
+            return False
+
+        return True
+
+print("")
+print("ILASM RoundTrips")
+
+if not os.path.exists("IL-RT"):
+  os.mkdir("IL-RT")
+
+if "arm" in platform.machine():
+  # TODO: Fix these roundtrip assemblies from failing on ARM64 when executing.
+  assembliesToIgnore = [
+    "value_numbering_checked_arithmetic_with_constants_ro.dll",
+    "ldc_mul_ovf_i4.dll",
+    "overflow02_mul.dll",
+    "b25815.dll",
+    "b71869.dll"
+  ]
+else:
+  assembliesToIgnore = []
+
+for inputAssemblyName in glob.glob("*.dll"):
+  if (not inputAssemblyName.endswith("$(AssemblyName).dll")) and is_managed_assembly(inputAssemblyName):
```

Comment:
Why is the `not` clause needed?  Isn't that the main dll for the test?

In a previous version, you had the list of files from the build system.  That seems simpler - what went wrong there?

Reply:
Yes, it is the main dll for the test. We still do the old ilasmroundtrip for 'exe' assemblies, and I do not want it to roundtrip a second time.

Reply:
Delete the old ilasmroundtrip code so that we don't have two copies of it.

Reply:
This will be hard because I don't know if the assembly is a library or exe when we do roundtripping.

Reply:
Maybe we don't care about the existing tests that use the old roundtripping?

Reply:
I'm not sure what you mean.  Most test project have a single binary and use old roundtripping, so we definitely care about running them.  If you mean specifically whether this PR's code should handle those binaries, we don't want to separate ways to run roundtripping.  That will confuse the debugging process and is difficult to maintain.

Reply:
It only makes it a little bit harder to maintain, but yea I would prefer if we only had one way. I just do not want to spend more time on this than I already have.

Reply:
It's hard to get it working because CI is so slow to verify.

I'll do the work to have only one impl.